### ISSUE TITLE
Involved destroying is NOT triggered

### ISSUE THREAD

**Issue by otiai10** [otiai10/awsub#67]
# as is

- many machines remain alive even though any of machines failed to be created

# to be

- all the machines are destroyed when any of machines failed to get created and executed

# facts

issued

```shell
SHARED_SPEC=m4.4xlarge
SAMPLE_COUNT=256
awsub \
    --script ${CFD}/bwa-mem.sh \
    --tasks ${CFD}/tasks/bwa-mem.${SAMPLE_COUNT}.csv \
    --image otiai10/bwa \
    --concurrency 32 \
    --shared REFERENCE=s3://awsub/resources/reference/GRCh37 \
    --env REFFILE=GRCh37.fa \
    --env CASE=${SHARED_SPEC}-x`printf %03d ${SAMPLE_COUNT}` \
    --aws-ec2-instance-type m4.large \
    --aws-iam-instance-profile awsubtest \
    --aws-shared-instance-type ${SHARED_SPEC} \
    --verbose \
    1>${CFD}/stdout.log 2>${CFD}/stderr.log ; echo $? >${CFD}/exitcode.log
```

`exitcode.log`

```
[ec2-user@ip-172-31-22-203 time-record]$ cat exitcode.log
1
```

`stderr.log`

```
[ec2-user@ip-172-31-22-203 time-record]$ cat stderr.log
2018/04/25 03:55:17 [COMMAND]	Your tasks file is parsed and decoded to 256 job(s) ✅
2018/04/25 03:55:17 [COMMAND]	See logs here -> /home/ec2-user/tmp/lab/projects/time-record/log/20180425_035517
2018/04/25 04:02:59 could not read CA certificate "/home/ec2-user/.docker/machine/machines/bwa-mem.256.csv-0104-m2flnwnm/ca.pem": open /home/ec2-user/.docker/machine/machines/bwa-mem.256.csv-0104-m2flnwnm/ca.pem: too many open files
```

docker-machine remained

```
[ec2-user@ip-172-31-22-203 ~]$ docker-machine ls
NAME                            ACTIVE   DRIVER      STATE     URL                         SWARM   DOCKER        ERRORS
bwa-mem.256.csv-0001-zddintg0   -        amazonec2   Running   tcp://54.250.143.211:2376           v18.04.0-ce
bwa-mem.256.csv-0002-y2zmndnl   -        amazonec2   Running   tcp://13.231.146.157:2376           v18.04.0-ce
bwa-mem.256.csv-0003-zmfjzdm1   -        amazonec2   Running   tcp://13.231.152.133:2376           v18.04.0-ce
bwa-mem.256.csv-0004-ngzizday   -        amazonec2   Running   tcp://13.231.144.117:2376           v18.04.0-ce
bwa-mem.256.csv-0005-mzhinmi4   -        amazonec2   Running   tcp://54.150.247.109:2376           v18.04.0-ce
bwa-mem.256.csv-0006-nwnjzmyx   -        amazonec2   Running   tcp://13.231.102.138:2376           v18.04.0-ce
bwa-mem.256.csv-0007-mmi5zjg4   -        amazonec2   Running   tcp://18.179.5.176:2376             v18.04.0-ce
bwa-mem.256.csv-0008-mjy0mtay   -        amazonec2   Running   tcp://13.113.212.55:2376            v18.04.0-ce
bwa-mem.256.csv-0009-njgyytgx   -        amazonec2   Running   tcp://52.198.5.203:2376             v18.04.0-ce
bwa-mem.256.csv-0010-zje2zwy0   -        amazonec2   Running   tcp://13.114.203.194:2376           v18.04.0-ce
bwa-mem.256.csv-0011-ode5m2fk   -        amazonec2   Running   tcp://54.64.206.175:2376            v18.04.0-ce
bwa-mem.256.csv-0012-zthjoda2   -        amazonec2   Running   tcp://175.41.234.16:2376            v18.04.0-ce
bwa-mem.256.csv-0013-otmwzgrj   -        amazonec2   Running   tcp://13.231.125.52:2376            v18.04.0-ce
bwa-mem.256.csv-0014-ymu3mta4   -        amazonec2   Running   tcp://13.112.243.54:2376            v18.04.0-ce
bwa-mem.256.csv-0015-mtzknjq2   -        amazonec2   Running   tcp://13.231.144.83:2376            v18.04.0-ce
bwa-mem.256.csv-0016-nwvmm2u4   -        amazonec2   Running   tcp://54.248.194.179:2376           v18.04.0-ce
bwa-mem.256.csv-0017-zjy0nty5   -        amazonec2   Running   tcp://13.231.123.18:2376            v18.04.0-ce
bwa-mem.256.csv-0018-mdizzguz   -        amazonec2   Running   tcp://54.238.229.225:2376           v18.04.0-ce
bwa-mem.256.csv-0019-ntqxodni   -        amazonec2   Running   tcp://54.238.250.238:2376           v18.04.0-ce
bwa-mem.256.csv-0020-mzdlotvh   -        amazonec2   Running   tcp://13.230.4.231:2376             v18.04.0-ce
bwa-mem.256.csv-0021-mde4mthm   -        amazonec2   Running   tcp://13.230.8.118:2376             v18.04.0-ce
bwa-mem.256.csv-0022-mtu5zwfk   -        amazonec2   Running   tcp://54.248.212.14:2376            v18.04.0-ce
bwa-mem.256.csv-0023-nzk2mgnk   -        amazonec2   Running   tcp://13.230.220.57:2376            v18.04.0-ce
bwa-mem.256.csv-0024-nmu5otbj   -        amazonec2   Running   tcp://52.196.181.233:2376           v18.04.0-ce
bwa-mem.256.csv-0025-ndfhmdhh   -        amazonec2   Running   tcp://13.230.107.69:2376            v18.04.0-ce
bwa-mem.256.csv-0026-odixzdfk   -        amazonec2   Running   tcp://13.231.156.74:2376            v18.04.0-ce
bwa-mem.256.csv-0027-ndjmmmjh   -        amazonec2   Running   tcp://13.114.30.92:2376             v18.04.0-ce
bwa-mem.256.csv-0028-mjuwngy1   -        amazonec2   Running   tcp://13.231.14.89:2376             v18.04.0-ce
bwa-mem.256.csv-0029-mgjkmtqz   -        amazonec2   Running   tcp://13.230.136.225:2376           v18.04.0-ce
bwa-mem.256.csv-0030-mzi0mzk5   -        amazonec2   Running   tcp://13.115.82.75:2376             v18.04.0-ce
bwa-mem.256.csv-0031-zdk4nmzi   -        amazonec2   Running   tcp://13.231.143.210:2376           v18.04.0-ce
bwa-mem.256.csv-0032-owqzywq4   -        amazonec2   Running   tcp://52.192.137.223:2376           v18.04.0-ce
bwa-mem.256.csv-0033-mwu2otzj   -        amazonec2   Running   tcp://54.95.145.127:2376            v18.04.0-ce
bwa-mem.256.csv-0034-yzrmotll   -        amazonec2   Running   tcp://54.249.132.74:2376            v18.04.0-ce
bwa-mem.256.csv-0035-nmyxotgw   -        amazonec2   Running   tcp://13.231.219.42:2376            v18.04.0-ce
bwa-mem.256.csv-0036-zmrhythh   -        amazonec2   Running   tcp://18.179.20.145:2376            v18.04.0-ce
bwa-mem.256.csv-0037-mjblzdgw   -        amazonec2   Running   tcp://54.249.135.230:2376           v18.04.0-ce
bwa-mem.256.csv-0038-zmy4mzi4   -        amazonec2   Running   tcp://54.95.43.8:2376               v18.04.0-ce
bwa-mem.256.csv-0039-ntg3mza4   -        amazonec2   Running   tcp://54.250.71.253:2376            v18.04.0-ce
bwa-mem.256.csv-0040-yzgwnte4   -        amazonec2   Running   tcp://13.230.68.87:2376             v18.04.0-ce
bwa-mem.256.csv-0041-mmy0odqw   -        amazonec2   Running   tcp://176.34.14.0:2376              v18.04.0-ce
bwa-mem.256.csv-0042-nji3zmu2   -        amazonec2   Running   tcp://13.113.194.20:2376            v18.04.0-ce
bwa-mem.256.csv-0043-ntjimgex   -        amazonec2   Running   tcp://54.250.45.188:2376            v18.04.0-ce
bwa-mem.256.csv-0044-ytvjotzj   -        amazonec2   Running   tcp://13.231.209.199:2376           v18.04.0-ce
bwa-mem.256.csv-0045-mgy2ymiz   -        amazonec2   Running   tcp://13.231.80.187:2376            v18.04.0-ce
bwa-mem.256.csv-0046-mge0ngnl   -        amazonec2   Running   tcp://13.231.109.104:2376           v18.04.0-ce
bwa-mem.256.csv-0047-ztywmjyz   -        amazonec2   Running   tcp://13.231.178.220:2376           v18.04.0-ce
bwa-mem.256.csv-0048-mjg1ndfl   -        amazonec2   Running   tcp://13.114.134.38:2376            v18.04.0-ce
bwa-mem.256.csv-0049-ywjjy2ix   -        amazonec2   Running   tcp://52.68.155.84:2376             v18.04.0-ce
bwa-mem.256.csv-0050-zddkmzi0   -        amazonec2   Running   tcp://54.150.203.253:2376           v18.04.0-ce
bwa-mem.256.csv-0051-ndjknzhm   -        amazonec2   Running   tcp://52.69.155.22:2376             v18.04.0-ce
bwa-mem.256.csv-0052-ndq0n2ji   -        amazonec2   Running   tcp://13.231.195.134:2376           v18.04.0-ce
bwa-mem.256.csv-0053-nji0y2e3   -        amazonec2   Running   tcp://52.196.219.64:2376            v18.04.0-ce
bwa-mem.256.csv-0054-otewyzfk   -        amazonec2   Running   tcp://52.193.207.64:2376            v18.04.0-ce
bwa-mem.256.csv-0055-zmi4mzky   -        amazonec2   Running   tcp://13.230.150.193:2376           v18.04.0-ce
bwa-mem.256.csv-0056-ownmzwy0   -        amazonec2   Running   tcp://13.230.110.183:2376           v18.04.0-ce
bwa-mem.256.csv-0057-n2q3mgzk   -        amazonec2   Running   tcp://52.196.156.168:2376           v18.04.0-ce
bwa-mem.256.csv-0058-odnln2jl   -        amazonec2   Running   tcp://52.197.136.159:2376           v18.04.0-ce
bwa-mem.256.csv-0059-mgzkzgyx   -        amazonec2   Running   tcp://13.231.137.230:2376           v18.04.0-ce
bwa-mem.256.csv-0060-ogezoty0   -        amazonec2   Running   tcp://54.64.204.211:2376            v18.04.0-ce
bwa-mem.256.csv-0061-ymziywvm   -        amazonec2   Running   tcp://13.114.91.127:2376            v18.04.0-ce
bwa-mem.256.csv-0062-m2iwzwm3   -        amazonec2   Running   tcp://13.231.37.226:2376            Unknown       Unable to query docker version: Cannot connect to the docker engine endpoint
bwa-mem.256.csv-0063-y2fhn2zi   -        amazonec2   Running   tcp://54.250.51.56:2376             v18.04.0-ce
bwa-mem.256.csv-0064-odfly2fi   -        amazonec2   Running   tcp://13.230.91.101:2376            Unknown       Unable to query docker version: Cannot connect to the docker engine endpoint
bwa-mem.256.csv-0065-ztzjntgw   -        amazonec2   Running   tcp://13.113.247.194:2376           Unknown       Unable to query docker version: Cannot connect to the docker engine endpoint
bwa-mem.256.csv-0066-yzzhnja0   -        amazonec2   Running   tcp://13.231.207.46:2376            Unknown       Unable to query docker version: Cannot connect to the docker engine endpoint
bwa-mem.256.csv-0067-ytnlnjk0   -        amazonec2   Running   tcp://13.231.211.80:2376            Unknown       Unable to query docker version: Cannot connect to the docker engine endpoint
bwa-mem.256.csv-0068-nzuwmwi5   -        amazonec2   Running   tcp://18.182.15.146:2376            Unknown       Unable to query docker version: Cannot connect to the docker engine endpoint
bwa-mem.256.csv-0069-mjqwm2i1   -        amazonec2   Running   tcp://13.230.90.219:2376            Unknown       Unable to query docker version: Cannot connect to the docker engine endpoint
bwa-mem.256.csv-0070-zmy5yzzj   -        amazonec2   Running   tcp://13.114.102.168:2376           Unknown       Unable to query docker version: Cannot connect to the docker engine endpoint
bwa-mem.256.csv-0071-y2m5mdlh   -        amazonec2   Running   tcp://18.179.10.251:2376            Unknown       Unable to query docker version: Cannot connect to the docker engine endpoint
bwa-mem.256.csv-0072-yzk4otez   -        amazonec2   Running   tcp://54.250.240.110:2376           v18.04.0-ce
bwa-mem.256.csv-0073-zmjjzjvj   -        amazonec2   Running   tcp://13.231.153.51:2376            v18.04.0-ce
bwa-mem.256.csv-0074-yjewntyy   -        amazonec2   Running   tcp://54.65.94.85:2376              v18.04.0-ce
bwa-mem.256.csv-0075-zda2mtni   -        amazonec2   Running   tcp://13.230.88.177:2376            v18.04.0-ce
bwa-mem.256.csv-0076-mddlndk3   -        amazonec2   Running   tcp://13.231.33.175:2376            v18.04.0-ce
bwa-mem.256.csv-0077-mji4zgm0   -        amazonec2   Running   tcp://54.199.212.35:2376            v18.04.0-ce
bwa-mem.256.csv-0078-mwm4mgvk   -        amazonec2   Running   tcp://54.178.150.57:2376            v18.04.0-ce
bwa-mem.256.csv-0079-y2eyntvk   -        amazonec2   Running   tcp://13.114.30.80:2376             v18.04.0-ce
bwa-mem.256.csv-0080-ngqxmdmw   -        amazonec2   Running   tcp://13.114.30.129:2376            v18.04.0-ce
bwa-mem.256.csv-0081-mwrjnwi2   -        amazonec2   Running   tcp://18.179.8.145:2376             v18.04.0-ce
bwa-mem.256.csv-0082-nda2yzcz   -        amazonec2   Running   tcp://13.114.103.14:2376            v18.04.0-ce
bwa-mem.256.csv-0083-yjk4yjhi   -        amazonec2   Running   tcp://13.114.103.9:2376             v18.04.0-ce
bwa-mem.256.csv-0084-m2e5nzm4   -        amazonec2   Running   tcp://13.231.51.141:2376            v18.04.0-ce
bwa-mem.256.csv-0085-ywq0mzq4   -        amazonec2   Running   tcp://13.230.212.189:2376           v18.04.0-ce
bwa-mem.256.csv-0086-zgqwyjax   -        amazonec2   Running   tcp://13.114.100.153:2376           v18.04.0-ce
bwa-mem.256.csv-0087-mja5nju0   -        amazonec2   Running   tcp://54.238.238.0:2376             v18.04.0-ce
bwa-mem.256.csv-0088-zmrizwe4   -        amazonec2   Running   tcp://13.115.136.98:2376            v18.04.0-ce
bwa-mem.256.csv-0089-mdvlnjzk   -        amazonec2   Running   tcp://13.113.162.47:2376            v18.04.0-ce
bwa-mem.256.csv-0090-nmjimdbh   -        amazonec2   Running   tcp://13.231.178.45:2376            v18.04.0-ce
bwa-mem.256.csv-0091-yzljmzky   -        amazonec2   Running   tcp://18.179.21.29:2376             v18.04.0-ce
bwa-mem.256.csv-0092-nzi4mjqz   -        amazonec2   Running   tcp://54.199.218.154:2376           v18.04.0-ce
bwa-mem.256.csv-0093-odbjzdli   -        amazonec2   Running   tcp://13.115.186.234:2376           v18.04.0-ce
bwa-mem.256.csv-0094-zdy5nte3   -        amazonec2   Running   tcp://54.95.163.103:2376            v18.04.0-ce
bwa-mem.256.csv-0095-njc4nmm2   -        amazonec2   Running   tcp://18.179.10.145:2376            v18.04.0-ce
bwa-mem.256.csv-0096-odhmnwu1   -        amazonec2   Running   tcp://175.41.215.82:2376            v18.04.0-ce
bwa-mem.256.csv-0097-zdrjmzi1   -        amazonec2   Running   tcp://13.231.233.6:2376             v18.04.0-ce
bwa-mem.256.csv-0098-n2vjzddi   -        amazonec2   Running   tcp://54.249.95.17:2376             v18.04.0-ce
bwa-mem.256.csv-0099-ntu5yjqx   -        amazonec2   Running   tcp://52.68.69.41:2376              v18.04.0-ce
bwa-mem.256.csv-0100-mdfjyzay   -        amazonec2   Running   tcp://18.182.5.242:2376             Unknown       Unable to query docker version: Cannot connect to the docker engine endpoint
bwa-mem.256.csv-0101-nji2zjq2   -        amazonec2   Running   tcp://18.179.16.141:2376            v18.04.0-ce
bwa-mem.256.csv-0102-zgzimgzj   -        amazonec2   Running   tcp://54.199.249.98:2376            Unknown       Unable to query docker version: Cannot connect to the docker engine endpoint
bwa-mem.256.csv-0103-yzy0yjyw   -        amazonec2   Running   tcp://13.231.242.9:2376             v18.04.0-ce
bwa-mem.256.csv-0104-m2flnwnm   -        amazonec2   Running   tcp://13.231.219.81:2376            v18.04.0-ce
bwa-mem.256.csv-0105-nmm4ztay   -        amazonec2   Running   tcp://18.179.23.151:2376            v18.04.0-ce
bwa-mem.256.csv-0106-zjy0zta5   -        amazonec2   Running   tcp://18.179.21.200:2376            v18.04.0-ce
bwa-mem.256.csv-0107-mdjknzq4   -        amazonec2   Running   tcp://13.230.91.138:2376            v18.04.0-ce
bwa-mem.256.csv-0108-ztgxodg5   -        amazonec2   Running   tcp://18.182.64.190:2376            Unknown       Unable to query docker version: Cannot connect to the docker engine endpoint
bwa-mem.256.csv-0109-ythlmwfh   -        amazonec2   Running   tcp://13.231.207.75:2376            v18.04.0-ce
bwa-mem.256.csv-0110-ndu4zwvj   -        amazonec2   Running   tcp://13.115.230.186:2376           Unknown       Unable to query docker version: Cannot connect to the docker engine endpoint
bwa-mem.256.csv-0111-zdmyzjlh   -        amazonec2   Running   tcp://13.231.72.157:2376            v18.04.0-ce
bwa-mem.256.csv-0112-mjlmmjnj   -        amazonec2   Running   tcp://54.248.70.2:2376              Unknown       Unable to query docker version: Cannot connect to the docker engine endpoint
bwa-mem.256.csv-0113-ngnhmzu3   -        amazonec2   Running   tcp://13.231.219.133:2376           v18.04.0-ce
bwa-mem.256.csv-0114-ndi2y2jl   -        amazonec2   Running   tcp://13.231.161.55:2376            Unknown       Unable to query docker version: Cannot connect to the docker engine endpoint
bwa-mem.256.csv-0115-mwuwnzzh   -        amazonec2   Running   tcp://18.182.12.165:2376            v18.04.0-ce
bwa-mem.256.csv-0116-ztgxmdg4   -        amazonec2   Running   tcp://13.230.184.226:2376           v18.04.0-ce
bwa-mem.256.csv-0117-njhmodjl   -        amazonec2   Running   tcp://13.114.141.172:2376           v18.04.0-ce
bwa-mem.256.csv-0118-ntfkotzj   -        amazonec2   Running   tcp://18.182.15.246:2376            v18.04.0-ce
bwa-mem.256.csv-0119-mjfjmjk4   -        amazonec2   Running   tcp://13.115.157.11:2376            v18.04.0-ce
bwa-mem.256.csv-0120-ntu2y2jk   -        amazonec2   Running   tcp://18.179.12.236:2376            Unknown       Unable to query docker version: Cannot connect to the docker engine endpoint
bwa-mem.256.csv-0121-ntyymgqw   -        amazonec2   Running   tcp://13.115.62.136:2376            Unknown       Unable to query docker version: Cannot connect to the docker engine endpoint
bwa-mem.256.csv-0122-mdjhzdzl   -        amazonec2   Running   tcp://13.115.98.134:2376            Unknown       Unable to query docker version: Cannot connect to the docker engine endpoint
bwa-mem.256.csv-0123-njq5nwuz   -        amazonec2   Running   tcp://54.238.249.174:2376           Unknown       Unable to query docker version: Cannot connect to the docker engine endpoint
bwa-mem.256.csv-0124-ngjmmtk3   -        amazonec2   Running   tcp://13.114.252.215:2376           v18.04.0-ce
bwa-mem.256.csv-0125-ywvhndy1   -        amazonec2   Running   tcp://13.114.118.235:2376           Unknown       Unable to query docker version: Cannot connect to the docker engine endpoint
bwa-mem.256.csv-0126-y2q4mdhk   -        amazonec2   Running   tcp://13.115.197.89:2376            Unknown       Unable to query docker version: Cannot connect to the docker engine endpoint
bwa-mem.256.csv-0127-yzbjmje2   -        amazonec2   Running   tcp://13.115.248.225:2376           Unknown       Unable to query docker version: Cannot connect to the docker engine endpoint
bwa-mem.256.csv-0128-mjq1mtvm   -        amazonec2   Running   tcp://13.115.78.249:2376            Unknown       Unable to query docker version: Cannot connect to the docker engine endpoint
bwa-mem.256.csv-0129-ymnjmzy1   -        amazonec2   Running   tcp://54.95.151.194:2376            Unknown       Unable to query docker version: Cannot connect to the docker engine endpoint
bwa-mem.256.csv-0130-m2jjzwy5   -        amazonec2   Running   tcp://13.231.248.213:2376           Unknown       Unable to query docker version: Cannot connect to the docker engine endpoint
bwa-mem.256.csv-0131-zwvizty1   -        amazonec2   Running   tcp://13.115.199.151:2376           Unknown       Unable to query docker version: Cannot connect to the docker engine endpoint
bwa-mem.256.csv-0132-ytu0njm4   -        amazonec2   Running   tcp://13.231.123.208:2376           Unknown       Unable to query docker version: Cannot connect to the docker engine endpoint
bwa-mem.256.csv-0133-yja1ntdh   -        amazonec2   Running   tcp://52.196.81.192:2376            Unknown       Unable to query docker version: Cannot connect to the docker engine endpoint
bwa-mem.256.csv-0134-ndu1mzji   -        amazonec2   Running   tcp://13.114.91.100:2376            Unknown       Unable to query docker version: Cannot connect to the docker engine endpoint
bwa-mem.256.csv-0135-nwrlzjg0   -        amazonec2   Running   tcp://18.182.9.201:2376             Unknown       Unable to query docker version: Cannot connect to the docker engine endpoint
bwa-mem.256.csv-0138-zjdmnjdm   -        amazonec2   Running   tcp://54.249.79.126:2376            Unknown       Unable to query docker version: Cannot connect to the docker engine endpoint
```

**Comment by otiai10** [comment]
# as is

- many machines remain alive even though any of machines failed to be created

# to be

- all the machines are destroyed when any of machines failed to be created

**Comment by otiai10** [comment]
seems it **was** triggered?????

```
[ec2-user@ip-172-31-22-203 time-record]$ cat stdout.log | grep DESTROY | wc -l
256
```

**Comment by otiai10** [comment]
```
[ec2-user@ip-172-31-22-203 time-record]$ cat ./stderr.log
2018/04/25 05:17:03 [COMMAND]	Your tasks file is parsed and decoded to 128 job(s) ✅
2018/04/25 05:17:03 [COMMAND]	See logs here -> /home/ec2-user/tmp/lab/projects/time-record/log/20180425_051703
2018/04/25 05:21:24 exit status 1: Running pre-create checks...
Creating machine...
(bwa-mem.128.csv-0009-y2m5zdk5) Launching instance...
Waiting for machine to be running, this may take a few minutes...
Detecting operating system of created instance...
Waiting for SSH to be available...
Detecting the provisioner...
Provisioning with ubuntu(systemd)...
Error creating machine: Error running provisioning: Error running "sudo apt-get update": ssh command error:
command : sudo apt-get update
err     : exit status 100
output  : Get:1 http://security.ubuntu.com/ubuntu xenial-security InRelease [107 kB]
Hit:2 http://archive.ubuntu.com/ubuntu xenial InRelease
Splitting up /var/lib/apt/lists/archive.ubuntu.com_ubuntu_dists_xenial_InRelease into data and signature failedErr:2 http://archive.ubuntu.com/ubuntu xenial InRelease
  Clearsigned file isn't valid, got 'NODATA' (does the network require authentication?)
Get:3 http://archive.ubuntu.com/ubuntu xenial-updates InRelease [109 kB]
Get:4 http://security.ubuntu.com/ubuntu xenial-security/main amd64 Packages [481 kB]
Get:5 http://archive.ubuntu.com/ubuntu xenial-backports InRelease [107 kB]
Get:6 http://security.ubuntu.com/ubuntu xenial-security/main Translation-en [207 kB]
Get:7 http://security.ubuntu.com/ubuntu xenial-security/restricted amd64 Packages [7,224 B]
Get:8 http://archive.ubuntu.com/ubuntu xenial-updates/main amd64 Packages [765 kB]
Get:9 http://security.ubuntu.com/ubuntu xenial-security/restricted Translation-en [2,152 B]
Get:10 http://security.ubuntu.com/ubuntu xenial-security/universe amd64 Packages [344 kB]
Get:11 http://security.ubuntu.com/ubuntu xenial-security/universe Translation-en [129 kB]
Get:12 http://security.ubuntu.com/ubuntu xenial-security/multiverse amd64 Packages [3,208 B]
Get:13 http://archive.ubuntu.com/ubuntu xenial-updates/main Translation-en [317 kB]
Get:14 http://security.ubuntu.com/ubuntu xenial-security/multiverse Translation-en [1,408 B]
Get:15 http://archive.ubuntu.com/ubuntu xenial-updates/restricted amd64 Packages [7,560 B]
Get:16 http://archive.ubuntu.com/ubuntu xenial-updates/restricted Translation-en [2,272 B]
Get:17 http://archive.ubuntu.com/ubuntu xenial-updates/universe amd64 Packages [623 kB]
Get:18 http://archive.ubuntu.com/ubuntu xenial-updates/universe Translation-en [251 kB]
Get:19 http://archive.ubuntu.com/ubuntu xenial-updates/multiverse amd64 Packages [16.2 kB]
Get:20 http://archive.ubuntu.com/ubuntu xenial-updates/multiverse Translation-en [8,076 B]
Get:21 http://archive.ubuntu.com/ubuntu xenial-backports/main amd64 Packages [4,836 B]
Get:22 http://archive.ubuntu.com/ubuntu xenial-backports/main Translation-en [3,220 B]
Get:23 http://archive.ubuntu.com/ubuntu xenial-backports/universe amd64 Packages [7,104 B]
Get:24 http://archive.ubuntu.com/ubuntu xenial-backports/universe Translation-en [3,844 B]
Reading package lists...
E: GPG error: http://archive.ubuntu.com/ubuntu xenial InRelease: Clearsigned file isn't valid, got 'NODATA' (does the network require authentication?)
```

```
[ec2-user@ip-172-31-22-203 time-record]$ cat stdout.log  | grep DESTROY | wc -l
128
```

**Comment by otiai10** [comment]
issued

```
[ec2-user@ip-172-31-22-203 time-record]$ ./driver.sh m4.10xlarge 64
awsub \
    --script ${CFD}/bwa-mem.sh \
    --tasks ${CFD}/tasks/bwa-mem.${SAMPLE_COUNT}.csv \
    --image otiai10/bwa \
    --concurrency 32 \
    --shared REFERENCE=s3://awsub/resources/reference/GRCh37 \
    --env REFFILE=GRCh37.fa \
    --env CASE=${SHARED_SPEC}-x`printf %03d ${SAMPLE_COUNT}` \
    --aws-ec2-instance-type m4.large \
    --aws-iam-instance-profile awsubtest \
    --aws-shared-instance-type ${SHARED_SPEC} \
    --verbose \
    1>${CFD}/stdout.log 2>${CFD}/stderr.log ; echo $? >${CFD}/exitcode.log
printf %03d ${SAMPLE_COUNT}

echo "DONE: `date`"
date
DONE: Wed Apr 25 07:34:39 UTC 2018
```

result


```
[ec2-user@ip-172-31-22-203 time-record]$ cat stderr.log
2018/04/25 07:30:51 [COMMAND]	Your tasks file is parsed and decoded to 64 job(s) ✅
2018/04/25 07:30:51 [COMMAND]	See logs here -> /home/ec2-user/tmp/lab/projects/time-record/log/20180425_073051
2018/04/25 07:34:39 exit status 1: Running pre-create checks...
Creating machine...
(bwa-mem.64.csv-0039-ody4otk1) Launching instance...
Waiting for machine to be running, this may take a few minutes...
Detecting operating system of created instance...
Waiting for SSH to be available...
Detecting the provisioner...
Provisioning with ubuntu(systemd)...
Error creating machine: Error running provisioning: Error running "sudo apt-get update": ssh command error:
command : sudo apt-get update
err     : exit status 100
output  : Get:1 http://security.ubuntu.com/ubuntu xenial-security InRelease [107 kB]
Hit:2 http://archive.ubuntu.com/ubuntu xenial InRelease
Splitting up /var/lib/apt/lists/archive.ubuntu.com_ubuntu_dists_xenial_InRelease into data and signature failedErr:2 http://archive.ubuntu.com/ubuntu xenial InRelease
  Clearsigned file isn't valid, got 'NODATA' (does the network require authentication?)
Get:3 http://archive.ubuntu.com/ubuntu xenial-updates InRelease [109 kB]
Get:4 http://security.ubuntu.com/ubuntu xenial-security/main amd64 Packages [481 kB]
Get:5 http://security.ubuntu.com/ubuntu xenial-security/main Translation-en [207 kB]
Get:6 http://archive.ubuntu.com/ubuntu xenial-backports InRelease [107 kB]
Get:7 http://security.ubuntu.com/ubuntu xenial-security/restricted amd64 Packages [7,224 B]
Get:8 http://security.ubuntu.com/ubuntu xenial-security/restricted Translation-en [2,152 B]
Get:9 http://archive.ubuntu.com/ubuntu xenial-updates/main amd64 Packages [765 kB]
Get:10 http://security.ubuntu.com/ubuntu xenial-security/universe amd64 Packages [344 kB]
Get:11 http://security.ubuntu.com/ubuntu xenial-security/universe Translation-en [129 kB]
Get:12 http://security.ubuntu.com/ubuntu xenial-security/multiverse amd64 Packages [3,208 B]
Get:13 http://security.ubuntu.com/ubuntu xenial-security/multiverse Translation-en [1,408 B]
Get:14 http://archive.ubuntu.com/ubuntu xenial-updates/main Translation-en [317 kB]
Get:15 http://archive.ubuntu.com/ubuntu xenial-updates/restricted amd64 Packages [7,560 B]
Get:16 http://archive.ubuntu.com/ubuntu xenial-updates/restricted Translation-en [2,272 B]
Get:17 http://archive.ubuntu.com/ubuntu xenial-updates/universe amd64 Packages [623 kB]
Get:18 http://archive.ubuntu.com/ubuntu xenial-updates/universe Translation-en [251 kB]
Get:19 http://archive.ubuntu.com/ubuntu xenial-updates/multiverse amd64 Packages [16.2 kB]
Get:20 http://archive.ubuntu.com/ubuntu xenial-updates/multiverse Translation-en [8,076 B]
Get:21 http://archive.ubuntu.com/ubuntu xenial-backports/main amd64 Packages [4,836 B]
Get:22 http://archive.ubuntu.com/ubuntu xenial-backports/main Translation-en [3,220 B]
Get:23 http://archive.ubuntu.com/ubuntu xenial-backports/universe amd64 Packages [7,104 B]
Get:24 http://archive.ubuntu.com/ubuntu xenial-backports/universe Translation-en [3,844 B]
Reading package lists...
E: GPG error: http://archive.ubuntu.com/ubuntu xenial InRelease: Clearsigned file isn't valid, got 'NODATA' (does the network require authentication?)

```

```
[ec2-user@ip-172-31-22-203 time-record]$ cat stdout.log  | grep DESTROY | wc -l
64
```

**Comment by otiai10** [comment]
issued

```
[ec2-user@ip-172-31-22-203 time-record]$ ./driver.sh m4.10xlarge 64
awsub \
    --script ${CFD}/bwa-mem.sh \
    --tasks ${CFD}/tasks/bwa-mem.${SAMPLE_COUNT}.csv \
    --image otiai10/bwa \
    --concurrency 32 \
    --shared REFERENCE=s3://awsub/resources/reference/GRCh37 \
    --env REFFILE=GRCh37.fa \
    --env CASE=${SHARED_SPEC}-x`printf %03d ${SAMPLE_COUNT}` \
    --aws-ec2-instance-type m4.large \
    --aws-iam-instance-profile awsubtest \
    --aws-shared-instance-type ${SHARED_SPEC} \
    --verbose \
    1>${CFD}/stdout.log 2>${CFD}/stderr.log ; echo $? >${CFD}/exitcode.log
printf %03d ${SAMPLE_COUNT}



echo "DONE: `date`"
date
DONE: Wed Apr 25 07:54:48 UTC 2018
```

result

```
[ec2-user@ip-172-31-22-203 time-record]$ cat exitcode.log
1
[ec2-user@ip-172-31-22-203 time-record]$ cat stderr.log
2018/04/25 07:47:24 [COMMAND]	Your tasks file is parsed and decoded to 64 job(s) ✅
2018/04/25 07:47:24 [COMMAND]	See logs here -> /home/ec2-user/tmp/lab/projects/time-record/log/20180425_074724
2018/04/25 07:54:48 Error: No such image: awsub/lifecycle
[ec2-user@ip-172-31-22-203 time-record]$ docker-machine ls -q | wc -l
0
```

# this means that

on creating or not created machines don't have `job.Instance` object on its `Job`, they can't even `Destroy` the machines.

**Comment by otiai10** [comment]
# cases: working well

```sh
[ec2-user@ip-172-31-22-203 time-record]$ cat stderr.log
2018/04/25 09:57:47 [COMMAND]	Your tasks file is parsed and decoded to 128 job(s) ✅
2018/04/25 09:57:47 [COMMAND]	See logs here -> /home/ec2-user/tmp/lab/projects/time-record/log/20180425_095747
2018/04/25 10:10:48 exit status 1: Running pre-create checks...
Creating machine...
(bwa-mem.128.csv-0027-oddkzdyx) Launching instance...
Waiting for machine to be running, this may take a few minutes...
Error creating machine: Error waiting for machine to be running: Maximum number of retries (60) exceeded

[ec2-user@ip-172-31-22-203 time-record]$ docker-machine ls -q | wc -l
0
```

**Comment by otiai10** [comment]
Fixed by f8ebb3fa124ae467f9c7ffe5b29b5a83325884e0
